### ISSUE TITLE
Wrong max extreme for the selection event in column charts

### ISSUE THREAD

**Issue by ollivih** [highcharts/highcharts#21058]
#### Expected behaviour
The events.selection callback function should have correct x-axis min and max values reflecting the selected range.

#### Actual behaviour
In the events.selection callback function the x-axis max is incorrect with some range selections. For example, with following screenshot selection, the x-axis max is around 1, but it should be around 0.5.
![image](https://github.com/highcharts/highcharts/assets/86189287/9d3d99f0-f689-40dd-8d13-64504347ab9c)

Seems that the bug appears when the selection range min is left side of the zero point. I can also see, that the min value doesn't go lower than 0 even though the x-axis left border is less than 0 (not sure if that is desired).

Assuming that the same issue is with the x-axis min value when selecting a range near the right border.

#### Live demo with steps to reproduce
Issue can be replicated with this demo (make a mouse selection shown in the above screenshot): https://jsfiddle.net/50Lq6jfh/

#### Product version
Highcharts v11.4.1

#### Affected browser(s)
Likely all browsers

I believe this issue is related to https://github.com/highcharts/highcharts/issues/20784.

**Comment by hubertkozik** [comment]
Hi @ollivih! Thank you for reporting the issue. The axis max value is correct, but the min value is wrong. This is a regression.

**Demo with axis min/max logging:** https://jsfiddle.net/BlackLabel/jbx4ydmh/

I've prepared a workaround for this issue. If you want to use it in your project just copy-paste the IIFE function to your project.

**Workaround:** https://jsfiddle.net/BlackLabel/asrfje0y/

_Internal note:_
Workaround is changing the `floor` and `ceiling` by adding/subtracting the `minPointOffset`.

**Comment by ollivih** [comment]
I feel that also the max value is incorrect with some selections.

For example in this selection, the max value seems correct (1):
![image](https://github.com/highcharts/highcharts/assets/86189287/662cae05-c829-435a-ad3c-1a9c67902016)

But in this selection, where the max is in the same place as in the previous, it gives 1.5 as the max value:
![image](https://github.com/highcharts/highcharts/assets/86189287/c47f8b8e-5c06-4e29-a9ac-1ed71faeb340)

**Comment by hubertkozik** [comment]
You're right @ollivih! My bad, sometimes the max values are incorrect also, but the workaround from my previous post is fixing max values also.
